src/mapDataManipulation/mapDataManipulationUtils.ts

```diff
@@ -24,6 +35,27 @@ export function changeRgbMappingArraysWithFunction(
 // [0,255] to [0,1]: a = 0, b = 255; c = 0, d = 1
 // [0,1] to [0,255]: a = 0, b = 1; c = 0, d = 255
 
+export function changeRgbMappingArrayInterval(
+  rgbMappingArray: number[],
+  oldMin: number,
+  oldMax: number,
+  newMin: number,
+  newMax: number,
+  strictlyLimitValuesToInterval: boolean = false,
+): number[] {
+  const transformValueToInterval = (x: number): number => {
+    let newX = newMin + ((newMax - newMin) / (oldMax - oldMin)) * (x - oldMin);
+    if (strictlyLimitValuesToInterval) {
+      newX = Math.max(newX, newMin);
+      newX = Math.min(newX, newMax);
+    }
+    return newX;
+  };
+
+  rgbMappingArray = changeRgbMappingArrayWithFunction(rgbMappingArray, transformValueToInterval);
+  return rgbMappingArray;
```

Comment:
`rgbMappingArray` (which comes from params) is being overwritten here. It doesn't matter much in this case, but maybe it is best to avoid it? I would suggest simply returning:
```
return changeRgbMappingArrayWithFunction(rgbMappingArray, transformValueToInterval);
```

Reply:
Thanks, did that, but also removed the usage of `changeRgbMappingArrayWithFunction` and now it's just

```javascript
return rgbMappingArray.map(x => transformValueToInterval(x));
```